src/Sylius/Bundle/CoreBundle/Controller/CheckoutController.php

```diff
@@ -193,6 +193,20 @@ public function finalizeAction(Request $request, OrderInterface $order)
     }
 
     /**
+     * @return Response
+     */
+    public function thankYouAction()
+    {
+        $id = $this->get('session')->get('sylius_order_id');
```

Comment:
Why if it is not set, we should throw a 404 as well, I do not remember but findOr404 should not work with ``null`` passed.